### PR TITLE
Fix product type taxes select

### PR DESCRIPTION
**Pull request by benekex2**
I want to merge this change because...

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/productTypes/views/ProductTypeCreate.tsx [hunk 1]
```diff
@@ -60,6 +60,13 @@ export const ProductTypeCreate: React.StatelessComponent = () => {
                     context: "page title"
                   })}
                   saveButtonBarState={loadingCreate ? "loading" : "default"}
+                  taxTypes={maybe(() => data.taxTypes, [
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/productTypes/views/ProductTypeCreate.tsx]: I believe that empty list would be sufficient

File: saleor/static/dashboard-next/productTypes/views/ProductTypeCreate.tsx [hunk 2]
```diff
@@ -60,6 +60,13 @@ export const ProductTypeCreate: React.StatelessComponent = () => {
                     context: "page title"
                   })}
                   saveButtonBarState={loadingCreate ? "loading" : "default"}
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/productTypes/views/ProductTypeCreate.tsx]: There is a helper function to do that, `getMutationState` from 'saleor/static/dashboard-next/misc.ts`

File: saleor/static/dashboard-next/productTypes/views/ProductTypeUpdate/index.tsx [hunk 3]
```diff
@@ -241,6 +241,13 @@ export const ProductTypeUpdate: React.StatelessComponent<
                         pageTitle={maybe(() => data.productType.name)}
                         productType={maybe(() => data.productType)}
                         saveButtonBarState={loading ? "loading" : "default"}
+                        taxTypes={maybe(() => data.taxTypes, [
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/productTypes/views/ProductTypeUpdate/index.tsx]: Same as above

**Comment by django-queries** [comment]
Here is the report for bef48dd23029cb3e5ad463e72cdffdcc49ca7261 (mirumee/saleor @ fix-product-type-taxes-select)
Base comparison is d254545366885804af3ddf5630ac9b8c24b8b949.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         43	         43	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

**Pull request merged**